### PR TITLE
Add the sprig functions in the template engine

### PR DESCRIPTION
**Pull request by thomasbach76**
### Description

This PR is the implementation of #1848. Basically it adds the sprig library (https://github.com/Masterminds/sprig) to the template engine.

### CODE REVIEW

File: build.Dockerfile [hunk 1]
```diff
@@ -12,7 +12,8 @@ RUN go get github.com/jteeuwen/go-bindata/... \
 && go get github.com/kisielk/errcheck \
 && go get github.com/client9/misspell/cmd/misspell \
 && go get github.com/mattfarina/glide-hash \
-&& go get github.com/sgotti/glide-vc
+&& go get github.com/sgotti/glide-vc \
+&& go get github.com/Masterminds/sprig
```
Comment (ldez) [inline build.Dockerfile]: Please remove this, we don't manage dependencies like that: https://github.com/containous/traefik/blob/master/.github/CONTRIBUTING.md#setting-up-glide-and-glide-vc-for-dependency-management

Comment (thomasbach76) [inline build.Dockerfile]: :ok_hand:

**Comment by ldez** [comment]
Could you do that:
1. Revert `glide.yml`, `glide.lock` and `vendor` changes. After that you must have nothing in the git stage.
1. Use the [github.com/multiplay/glide-pin](https://github.com/multiplay/glide-pin) plugin: `glide pin --exact`
1. Commit the `glide.yml` with pinned version
1. Add dependency: `./script/glide.sh get YOUR/DEPENDENCY`
1. Commit `glide.yml`, `glide.lock` and `vendor`
1. Revert the pin commit: `git revert HASH_OF_PIN_COMMIT`
1. Edit manually `glide.yml` to add the dependency
1. Use [glide-hash](https://github.com/mattfarina/glide-hash) to get the manifest file hash and put it in the lock file to please `validate-glide`.
1. Commit `glide.yml` and `glide.lock`
1. Squash all previous commit: `git rebase -i HASH`

**Comment by ldez** [comment]
please follow exactly my previous comment:
- first, revert all your changes on `glide.lock`, `glide.yml` and vendor (step 1)

**Comment by emilevauge** [comment]
Hi @thomasbach76, thanks for your great contribution!
Design LGTM

**Comment by thomasbach76** [comment]
Not sure why the semaphoreci check failed here...

**Comment by ldez** [comment]
It's not you, I have relaunch the build.

**Pull request merged**